src/neomaril_codex/model.py

```diff
@@ -322,42 +340,97 @@ def delete(self):
         Returns
         -------
         str
-            If model is at status=Deployed deletes the model and return a json with his information. If it isn't Deployed it returns the message that the model is under another state
+            If model is at status=Deployed deletes the model and return a json with his information.
+            If it isn't Deployed it returns the message that the model is under another state
 
         Example
         -------
         >>> model.delete()
         """
-        if self.__model_ready:
-            token = refresh_token(*self.credentials, self.base_url)
-            req = requests.delete(
-                f"{self.base_url}/model/delete/{self.group}/{self.model_id}",
-                headers={"Authorization": "Bearer " + token},
-            )
+        if self.__get_status() not in [ModelState.Disabled, ModelState.DisabledFailed, ModelState.Failed]:
+            return "Cannot delete the model! Current state is " + str(self.status)
 
-            if req.status_code == 200:
-                response = requests.get(
-                    f"{self.base_url}/model/describe/{self.group}/{self.model_id}",
-                    headers={"Authorization": "Bearer " + token},
-                )
+        token = refresh_token(*self.credentials, self.base_url)
+        req = requests.delete(
+            f"{self.base_url}/model/delete/{self.group}/{self.model_id}",
+            headers={"Authorization": "Bearer " + token},
+        )
 
-                self.model_data = response.json()["Description"]
-                self.status = self.model_data["Status"]
-                self.__model_ready = False
+        if req.status_code == 401:
+            logger.error(req.text)
+            raise AuthenticationError("Login not authorized")
 
-                return req.json()
-            elif req.status_code == 401:
-                logger.error(req.text)
-                raise AuthenticationError("Login not authorized")
-            elif req.status_code >= 500:
-                logger.error(req.text)
-                raise ServerError("Server Error")
-            else:
-                logger.error(req.text)
-                raise ModelError("Could not delete the model")
+        if req.status_code >= 500:
+            logger.error(req.text)
+            raise ServerError("Server Error")
 
-        else:
-            return "Model is " + self.status
+        if req.status_code != 200:
+            logger.error(req.text)
+            raise ModelError("Failed to delete model.")
+
+        response = requests.get(
+            f"{self.base_url}/model/describe/{self.group}/{self.model_id}",
+            headers={"Authorization": "Bearer " + token},
+        )
+
+        self.model_data = response.json()["Description"]
+        self.status = ModelState[self.model_data["Status"]]
+        self.__model_ready = False
+
+        return req.json()
+
+    def disable(self):
```

Comment:
What do you think about adding this function to some sample like `Model.ipynb`?

Reply:
https://github.com/datarisk-io/mlops-neomaril-codex/pull/50/commits/e78d597e34fd29caae7342b0bbb7a6abf2e00dbd